sdk/src/metrics/export/periodic_exporting_metric_reader.cc

```diff
@@ -111,7 +118,7 @@ bool PeriodicExportingMetricReader::CollectAndExportOnce()
     status = future_receive.wait_for(std::chrono::milliseconds(export_timeout_millis_));
     if (status == std::future_status::timeout)
     {
-      cancel_export_for_timeout = true;
+      cancel_export_for_timeout->store(true, std::memory_order_release);
```

Comment:
Independently of this PR (to discuss):

* The ``::Collect()` loop processes all instruments at once today.
* We could pass the cancel_export_for_timeout pointer to the collect method, so collect can check at each loop if the timeout already is passed.
* This will make collect more responsive, and abort sooner when a lot of metrics are defined.

Also:

* shutdown may need to signal to pending exports to stop.
* For this, the class `PeriodicExportingMetricReader` can keep a reference to the cancel_export_for_timeout pointers, and change them on shutdown, to notify exporter threads to stop

This will increase robustness, and prevent too many threads to run concurrently.

Reply:
Good point. But can we implement it in another PR? This is just a quick fix for crash.

Reply:
Yes, definitively in another PR.